src/services/divisionService.js

```diff
@@ -0,0 +1,153 @@
+const {
+  addDivision,
+  getAllDivisions,
+  getDivisionBySlug,
+  updateDivisionBySlug,
+  deleteDivisionBySlug,
+} = require('../models/divisionModel');
+const { createSlugDivision } = require('../helpers/slug');
+const {
+  getUploadFilepath,
+  updateFileDivision,
+} = require('../utils/uploadFileDivision');
+
+/**
+ * Checks if 'slug' index exists in the database.
+ *
+ * @param {*} slug
+ * @returns
+ */
+const checkSlugExistsInDb = async (slug) => {
+  const useCache = false;
+  const division = await getDivisionBySlug(slug, useCache);
+  return !!division;
+};
+exports.checkSlugExistsInDb = checkSlugExistsInDb;
+
+/**
+ * Add new division into the database.
+ *
+ * @param {*} data
+ * @returns
+ */
+exports.addDivision = async (data) => {
+  try {
+    return addDivision(data);
+  } catch (error) {
+    console.error('Error adding divisions:', error);
+    throw error;
+  }
+};
```

Comment:
Consider using 'await' for the promise inside this 'try' or replace it with 'Promise.prototype.catch(...)' usage. Promise rejections should not be caught by "try" blocks.

## Why this is an issue?
An exception thrown inside a promise will not be caught by a nesting try block due to the asynchronous nature of execution.
Promises are designed to propagate errors to the next error handler or catch() block in the promise chain. Promises are asynchronous and operate outside of the normal call stack. When a Promise is created, it is added to the microtask queue, which is processed after the current call stack has completed. This means that the try-catch block surrounding the Promise will have already completed by the time the Promise is resolved or rejected. Therefore, any error occurring within the Promise will not be caught by the try-catch block.

## Software qualities impacted
Issues found for this rule will have a medium impact on the reliability of your software.

## Resources
You can read the following resources for references
- MDN web docs - [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
- MDN web docs - [try...catch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch)
- MDN web docs - [await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await)
- MDN web docs - [async function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function)
- MDN web docs - [Microtasks](https://developer.mozilla.org/en-US/docs/Web/API/HTML_DOM_API/Microtask_guide)